privacyidea/static/components/components/controllers/componentControllers.js

```diff
@@ -42,6 +44,16 @@ myApp.controller("componentController", function (ComponentFactory, $scope,
         $location.path("/component/clienttype");
     }
 
+    $scope.uploadAsk = function(files) {
```

Comment:
Is it necessary to add an extra function f for each call?
I think we might produce less code, when we add the "confirm" function directly into the html template like 

~~~~javascript
        <button
                class="btn btn-primary"
                ng-file-select
                ng-file-change='confirm("difficult", 
                                          {{"Upload Subscription File" | translate}},
                                          {{"Do you want to import the selected file?" | translate}},
                                          {{"Start Upload" | translate}},
                                          $scope.upload($files))' 
                translate>
            Upload Subscription File
        </button>
~~~~

This way

1. we could have much less functions / code
2. have labeling of the button *and* the confirmation texts in one place
3. Also have the confirmation level togeather with the template
4. have all the translation in one place

This might look a bit more complicated, but I think it is really awkward to have to many wrapper functions.
What do you think?